### PR TITLE
[ADD] odoo - unitest - output test results as XML

### PR DESCRIPTION
**Pull request by Jason-Scott-23**
Description of the issue/feature this PR addresses: XML output of unit tests

Current behavior before PR: No output of unit test results

Desired behavior after PR is merged: unit tests will have XML outputs

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/modules/loading.py [hunk 1]
```diff
@@ -261,14 +262,21 @@ def load_module_graph(cr, graph, status=None, perform_checks=True,
...
                 tests_t0, tests_q0 = time.time(), odoo.sql_db.sql_counter
-                test_results = loader.run_suite(suite, module_name)
-                report.update(test_results)
-                test_time = time.time() - tests_t0
-                test_queries = odoo.sql_db.sql_counter - tests_q0
-
-                # tests may have reset the environment
-                env = api.Environment(cr, SUPERUSER_ID, {})
-                module = env['ir.module.module'].browse(module_id)
+                # test_results = loader.run_suite(suite, module_name)
```
Comment (kdwyer) [inline odoo/modules/loading.py]: I think this line should be uncommented and moved into an `else:` block on the following `if`?

File: odoo/modules/loading.py [hunk 2]
```diff
@@ -261,14 +262,21 @@ def load_module_graph(cr, graph, status=None, perform_checks=True,
...
+                if config['test_xml_file']:
+                    import xmlrunner
+                    xml_output_file_name = '.'.join([config['test_xml_file'], str(time.time()), 'xml'])
+                    _logger.info('saving xml results to %s', xml_output_file_name)
+                    xml_output_file = open(xml_output_file_name, 'wb')
+                    runner = xmlrunner.XMLTestRunner(output=xml_output_file)
+                    test_results = runner.run(suite)
+                    report.update(test_results)
```
Comment (kdwyer) [inline odoo/modules/loading.py]: This line and those following should be dedented, so that we can run the tests without specifying xml output?

**Review by kdwyer — APPROVED** [review]

File: odoo/modules/loading.py [hunk 3]
```diff
@@ -261,7 +262,16 @@ def load_module_graph(cr, graph, status=None, perform_checks=True,
                 env['ir.http']._clear_routing_map()     # force routing map to be rebuilt
 
                 tests_t0, tests_q0 = time.time(), odoo.sql_db.sql_counter
-                test_results = loader.run_suite(suite, module_name)
+                if config['test_xml_file']:
```
Comment (williamsajr) [inline odoo/modules/loading.py]: It may be safer to do `if config.get('test_xml_file'):`

Comment (Jason-Scott-23) [reply odoo/modules/loading.py]: I've updated it - thanks

**Review by williamsajr — APPROVED** [review]

**Pull request merged**